ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -197,24 +275,26 @@ instance ToCBOR VRFType where
   -- This is a cheat, and at some point we probably want to decide on Serialise/ToCBOR
   toCBOR = encode
 
-deriving instance PraosCrypto c => Show (PraosExtraFields c)
-deriving instance PraosCrypto c => Eq   (PraosExtraFields c)
-
-deriving instance PraosCrypto c => Show (PraosFields c toSign)
-deriving instance PraosCrypto c => Eq   (PraosFields c toSign)
-
+-- |Proofs certifying ρ and y for a given slot and eta.
 data PraosProof c = PraosProof {
       praosProofRho :: CertifiedVRF (PraosVRF c) (Natural, SlotNo, VRFType)
     , praosProofY   :: CertifiedVRF (PraosVRF c) (Natural, SlotNo, VRFType)
     , praosLeader   :: CoreNodeId
     }
 
+-- | An error that can arise during validation
 data PraosValidationError c =
+      -- | The slot was invalid because it is not strictly increasing from the
+      -- previous block
       PraosInvalidSlot SlotNo SlotNo
-    | PraosUnknownCoreId CoreNodeId
-    | PraosInvalidSig String (VerKeyKES (PraosKES c)) Natural (SigKES (PraosKES c))
-    | PraosInvalidCert (VerKeyVRF (PraosVRF c)) (Natural, SlotNo, VRFType) Natural (CertVRF (PraosVRF c))
-    | PraosInsufficientStake Double Natural
+    | -- | The author of the block is unknown
```

Comment:
Style: we put the `|` on the line with the constructor instead of on the line with the `-- |`. See https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/docs/StyleGuide.md (CTRL-F for "-- | Baar".)

Reply:
Right, I actually removed these comments. But thanks for the pointer to the style guide.